unix/w0vncserver/wayland/WaylandDesktop.cxx

```diff
@@ -129,6 +133,14 @@ void WaylandDesktop::pointerEvent(const core::Point& pos, uint16_t buttonMask)
 
 void WaylandDesktop::keyEvent(uint32_t keysym, uint32_t keycode, bool down)
 {
+  if (rawKeyboard) {
+    if (keycode >= code_map_qnum_to_xorgevdev_len) {
+      vlog.error("Could not map key event to evdev key code");
+      return;
+    }
+    keycode = code_map_qnum_to_xorgevdev[keycode];
```

Comment:
I don't think this is correct. You have a layout here, so it's not a given that evdev key codes are used. You should probably do what vncviewer does to map things.